pkg/clusterversion/cv.go

```diff
@@ -289,8 +289,8 @@ func (c *clusterVersionClient) runUpgradeWithChannelVersion(cv *configv1.Cluster
 
 	if cv.Spec.Channel != desired.Channel {
 		logger.Info(fmt.Sprintf("Setting ClusterVersion to Channel %s Version %s", desired.Channel, desired.Version))
-		cv.Spec.Channel = desired.Channel
-		err := c.client.Update(context.TODO(), cv)
+		patch := []byte(fmt.Sprintf(`{"spec":{"channel":"%s"}}`, desired.Channel))
```

Comment:
As you've encountered during testing, having the hand-written json strings can lead to inadvertent test failures simply due to things such as an extra space character, etc, which is a little bit ugly.

The simplest way would be to switch the patch to using an approach via `StrategicMergeFrom` ([link](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client#StrategicMergeFrom)) which generates the patch for you by comparing two objects.

I'd wondered if building up the update using the [struct](https://github.com/openshift/api/blob/master/config/v1/types_cluster_version.go#L364) and then marshalling that to a json string would work, but that would not result in a correct patch (it'd be missing the outer `spec` and `desiredUpdate` layers).

But that would be pretty much re-work of the code and tests to switch to the `StrategicMergeFrom` approach. Maybe you could write it up as a low-hanging-fruit toil maintenance ticket to look at in the future.

Reply:
I am trying to work out how to switch to using `MergeFrom` and what needs to change in the tests to get those passing.
I am not sure if I will be finished that today.
I am also wondering if that would be best done in it's own change.